src/app/tasks/services/tasks-filters.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+    }
+  };
+  const mockTableService = {
+    saveFilters: jest.fn(),
+    resetFilters: jest.fn(),
+    restoreFilters: jest.fn(),
+  };
+  const mockFiltersDefinitions : TaskFilterDefinition[] = [
+    {
+      for: 'root',
+      field: 16,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 1,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 10,
+      type: 'string',
+    },
+    {
+      for: 'root',
+      field: 2,
+      type: 'status',
+      statuses: Object.keys(mockTasksStatusesService.statuses).map(status => {
+        return {
+          key: status,
+          value: mockTasksStatusesService.statuses[Number(status) as TaskStatus],
+        };
+      }),
+    },
+    {
+      for: 'options',
+      field: 5,
+      type: 'string',
+    },
+    {
+      for: 'options',
+      field: 6,
+      type: 'string',
+    },
+    {
+      for: 'options',
+      field: 4,
+      type: 'string',
+    }
+  ];
+
+  const filterAnd: FiltersAnd<number, number> = [{
+    field: 1,
+    for: 'root',
+    operator: 1,
+    value: 'Dummy'
+  }];
+  const mockTaskSummaryfilterOr: TaskSummaryFiltersOr = [
+    filterAnd
+  ];
+
+  beforeEach(() => {
+    service = TestBed.configureTestingModule({
+      providers: [
+        TasksFiltersService,
+        {provide: TasksStatusesService, useValue: mockTasksStatusesService},
+        {provide: DefaultConfigService, useValue: mockDefaultConfigService }, 
+        {provide: TableService, useValue: mockTableService}
+        
+      ]
+    }).inject(TasksFiltersService);
+  });
+
+  test('the service must create TasksFilterService', () => {
+    expect(service).toBeTruthy();
+  }); 
+
+  test('the service must call saveFilters from Table Service', () => {
+    const filters = mockTaskSummaryfilterOr;
+    service.saveFilters(filters);
+    const spySaveFilters = jest.spyOn(mockTableService, 'saveFilters');
+    expect(spySaveFilters).toHaveBeenCalled();
+    expect(spySaveFilters).toHaveBeenCalledWith('tasks-filters', filters);
+  });
+
+  test('the service must call restoreFilters from Table Service', () => {
+    service.restoreFilters();
+    const spyRestoreFilters = jest.spyOn(mockTableService, 'restoreFilters');
+    expect(spyRestoreFilters).toHaveBeenCalled();
+    expect(spyRestoreFilters).toHaveBeenCalledWith('tasks-filters', mockFiltersDefinitions);
+  });
+  test('the service must call resetFilters from Table Service', () => {
+    service.resetFilters();
+    const spyRestoreFilters = jest.spyOn(mockTableService, 'resetFilters');
+    const mockDefaultFilters = mockDefaultConfigService.defaultTasks.filters;
+    expect(spyRestoreFilters).toHaveBeenCalled();
+    expect(spyRestoreFilters).toHaveBeenCalledWith('tasks-filters');
+    expect(service.resetFilters()).toBe(mockDefaultFilters);
+  });
+  test('the service must return #filtersDefinitions', () =>{
+    expect(service.retrieveFiltersDefinitions()).toEqual(mockFiltersDefinitions);
```

Comment:
As i said in a previous comment, the filtersDefinition is not supposed to be mocked. It can be used as an expected value, but don't call it a mock.

```suggestion
    expect(service.retrieveFiltersDefinitions()).toEqual(expectedFiltersDefinitions);
```